processing/src/main/java/io/druid/segment/data/GenericIndexed.java

```diff
@@ -272,6 +283,43 @@ public int indexOf(T value)
     }
   }
 
+  class BufferIndexedV2 extends BufferIndexed {
+
+    @Override
+    protected T _get(final ByteBuffer copyBuffer, final int index) {
+      if (index < 0) {
+        throw new IAE("Index[%s] < 0", index);
+      }
+      if (index >= size) {
+        throw new IAE(String.format("Index[%s] >= size[%s]", index, size));
+      }
+
+      final int startOffset;
+      final int endOffset;
+      final int valueLengthDelta = 0;
```

Comment:
I think that should be doable by passing in this constant to the constructor of `BufferIndexed` class